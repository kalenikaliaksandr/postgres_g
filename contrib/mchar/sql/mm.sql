select 'asd'::mchar::mvarchar;
select 'asd '::mchar::mvarchar;
select 'asd'::mchar(2)::mvarchar;
select 'asd '::mchar(2)::mvarchar;
select 'asd'::mchar(5)::mvarchar;
select 'asd '::mchar(5)::mvarchar;
select 'asd'::mchar::mvarchar(2);
select 'asd '::mchar::mvarchar(2);
select 'asd'::mchar(2)::mvarchar(2);
select 'asd '::mchar(2)::mvarchar(2);
select 'asd'::mchar(5)::mvarchar(2);
select 'asd '::mchar(5)::mvarchar(2);
select 'asd'::mchar::mvarchar(5);
select 'asd '::mchar::mvarchar(5);
select 'asd'::mchar(2)::mvarchar(5);
select 'asd '::mchar(2)::mvarchar(5);
select 'asd'::mchar(5)::mvarchar(5);
select 'asd '::mchar(5)::mvarchar(5);

select 'asd'::mvarchar::mchar;
select 'asd '::mvarchar::mchar;
select 'asd'::mvarchar(2)::mchar;
select 'asd '::mvarchar(2)::mchar;
select 'asd'::mvarchar(5)::mchar;
select 'asd '::mvarchar(5)::mchar;
select 'asd'::mvarchar::mchar(2);
select 'asd '::mvarchar::mchar(2);
select 'asd'::mvarchar(2)::mchar(2);
select 'asd '::mvarchar(2)::mchar(2);
select 'asd'::mvarchar(5)::mchar(2);
select 'asd '::mvarchar(5)::mchar(2);
select 'asd'::mvarchar::mchar(5);
select 'asd '::mvarchar::mchar(5);
select 'asd'::mvarchar(2)::mchar(5);
select 'asd '::mvarchar(2)::mchar(5);
select 'asd'::mvarchar(5)::mchar(5);
select 'asd '::mvarchar(5)::mchar(5);

select 'asd'::mchar || '123';
select 'asd'::mchar || '123'::mchar;
select 'asd'::mchar || '123'::mvarchar;

select 'asd '::mchar || '123';
select 'asd '::mchar || '123'::mchar;
select 'asd '::mchar || '123'::mvarchar;

select 'asd '::mchar || '123 ';
select 'asd '::mchar || '123 '::mchar;
select 'asd '::mchar || '123 '::mvarchar;


select 'asd'::mvarchar || '123';
select 'asd'::mvarchar || '123'::mchar;
select 'asd'::mvarchar || '123'::mvarchar;

select 'asd '::mvarchar || '123';
select 'asd '::mvarchar || '123'::mchar;
select 'asd '::mvarchar || '123'::mvarchar;

select 'asd '::mvarchar || '123 ';
select 'asd '::mvarchar || '123 '::mchar;
select 'asd '::mvarchar || '123 '::mvarchar;


select 'asd'::mchar(2) || '123';
select 'asd'::mchar(2) || '123'::mchar;
select 'asd'::mchar(2) || '123'::mvarchar;


select 'asd '::mchar(2) || '123';
select 'asd '::mchar(2) || '123'::mchar;
select 'asd '::mchar(2) || '123'::mvarchar;


select 'asd '::mchar(2) || '123 ';
select 'asd '::mchar(2) || '123 '::mchar;
select 'asd '::mchar(2) || '123 '::mvarchar;

select 'asd'::mvarchar(2) || '123';
select 'asd'::mvarchar(2) || '123'::mchar;
select 'asd'::mvarchar(2) || '123'::mvarchar;

select 'asd '::mvarchar(2) || '123';
select 'asd '::mvarchar(2) || '123'::mchar;
select 'asd '::mvarchar(2) || '123'::mvarchar;

select 'asd '::mvarchar(2) || '123 ';
select 'asd '::mvarchar(2) || '123 '::mchar;
select 'asd '::mvarchar(2) || '123 '::mvarchar;

select 'asd'::mchar(4) || '143';
select 'asd'::mchar(4) || '123'::mchar;
select 'asd'::mchar(4) || '123'::mvarchar;

select 'asd '::mchar(4) || '123';
select 'asd '::mchar(4) || '123'::mchar;
select 'asd '::mchar(4) || '123'::mvarchar;

select 'asd '::mchar(4) || '123 ';
select 'asd '::mchar(4) || '123 '::mchar;
select 'asd '::mchar(4) || '123 '::mvarchar;

select 'asd'::mvarchar(4) || '123';
select 'asd'::mvarchar(4) || '123'::mchar;
select 'asd'::mvarchar(4) || '123'::mvarchar;

select 'asd '::mvarchar(4) || '123';
select 'asd '::mvarchar(4) || '123'::mchar;
select 'asd '::mvarchar(4) || '123'::mvarchar;

select 'asd '::mvarchar(4) || '123 ';
select 'asd '::mvarchar(4) || '123 '::mchar;
select 'asd '::mvarchar(4) || '123 '::mvarchar;


select 'asd '::mvarchar(4) || '123 '::mchar(4);
select 'asd '::mvarchar(4) || '123 '::mvarchar(4);
select 'asd '::mvarchar(4) || '123'::mchar(4);
select 'asd '::mvarchar(4) || '123'::mvarchar(4);


select 1 where 'f'::mchar='F'::mvarchar;
select 1 where 'f'::mchar='F '::mvarchar;
select 1 where 'f '::mchar='F'::mvarchar;
select 1 where 'f '::mchar='F '::mvarchar;

select 1 where 'f'::mchar='F'::mvarchar(2);
select 1 where 'f'::mchar='F '::mvarchar(2);
select 1 where 'f '::mchar='F'::mvarchar(2);
select 1 where 'f '::mchar='F '::mvarchar(2);

select 1 where 'f'::mchar(2)='F'::mvarchar;
select 1 where 'f'::mchar(2)='F '::mvarchar;
select 1 where 'f '::mchar(2)='F'::mvarchar;
select 1 where 'f '::mchar(2)='F '::mvarchar;

select 1 where 'f'::mchar(2)='F'::mvarchar(2);
select 1 where 'f'::mchar(2)='F '::mvarchar(2);
select 1 where 'f '::mchar(2)='F'::mvarchar(2);
select 1 where 'f '::mchar(2)='F '::mvarchar(2);

select 1 where 'foo'::mchar='FOO'::mvarchar;
select 1 where 'foo'::mchar='FOO '::mvarchar;
select 1 where 'foo '::mchar='FOO'::mvarchar;
select 1 where 'foo '::mchar='FOO '::mvarchar;

select 1 where 'foo'::mchar='FOO'::mvarchar(2);
select 1 where 'foo'::mchar='FOO '::mvarchar(2);
select 1 where 'foo '::mchar='FOO'::mvarchar(2);
select 1 where 'foo '::mchar='FOO '::mvarchar(2);

select 1 where 'foo'::mchar(2)='FOO'::mvarchar;
select 1 where 'foo'::mchar(2)='FOO '::mvarchar;
select 1 where 'foo '::mchar(2)='FOO'::mvarchar;
select 1 where 'foo '::mchar(2)='FOO '::mvarchar;

select 1 where 'foo'::mchar(2)='FOO'::mvarchar(2);
select 1 where 'foo'::mchar(2)='FOO '::mvarchar(2);
select 1 where 'foo '::mchar(2)='FOO'::mvarchar(2);
select 1 where 'foo '::mchar(2)='FOO '::mvarchar(2);

Select 'f'::mchar(1) Union Select 'o'::mvarchar(1);
Select 'f'::mvarchar(1) Union Select 'o'::mchar(1);

select * from chvch where ch=vch;

select ch.* from ch, (select 'dEfg'::mvarchar as q) as p  where  chcol > p.q;
create index qq on ch (chcol);
set enable_seqscan=off;
select ch.* from ch, (select 'dEfg'::mvarchar as q) as p  where  chcol > p.q;
set enable_seqscan=on;


--\copy chvch to 'results/chvch.dump' binary
--truncate table chvch;
--\copy chvch from 'results/chvch.dump' binary

--test joins
CREATE TABLE a (mchar2 MCHAR(2) NOT NULL);
CREATE TABLE c (mvarchar255 mvarchar NOT NULL);
SELECT * FROM a, c WHERE mchar2 = mvarchar255;
SELECT * FROM a, c WHERE mvarchar255 = mchar2;
DROP TABLE a;
DROP TABLE c;

select * from (values
    ('е'::mchar),('ё'),('еа'),('еб'),('ее'),('еж'),('ёа'),('ёб'),('ёё'),('ёж'),('ёе'),('её'))
    z order by 1;

select 'ё'::mchar = 'е';
select 'Ё'::mchar = 'Е';
select 'й'::mchar = 'и';
select 'Й'::mchar = 'И';

select mvarchar_icase_cmp('ёа','еб'), mvarchar_icase_cmp('еб','ё'),
    mvarchar_icase_cmp('ё', 'ёа');
