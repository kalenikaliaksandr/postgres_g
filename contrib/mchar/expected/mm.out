select 'asd'::mchar::mvarchar;
 mvarchar 
----------
 asd
(1 row)

select 'asd '::mchar::mvarchar;
 mvarchar 
----------
 asd
(1 row)

select 'asd'::mchar(2)::mvarchar;
 mvarchar 
----------
 as
(1 row)

select 'asd '::mchar(2)::mvarchar;
 mvarchar 
----------
 as
(1 row)

select 'asd'::mchar(5)::mvarchar;
 mvarchar 
----------
 asd  
(1 row)

select 'asd '::mchar(5)::mvarchar;
 mvarchar 
----------
 asd  
(1 row)

select 'asd'::mchar::mvarchar(2);
 mvarchar 
----------
 as
(1 row)

select 'asd '::mchar::mvarchar(2);
 mvarchar 
----------
 as
(1 row)

select 'asd'::mchar(2)::mvarchar(2);
 mvarchar 
----------
 as
(1 row)

select 'asd '::mchar(2)::mvarchar(2);
 mvarchar 
----------
 as
(1 row)

select 'asd'::mchar(5)::mvarchar(2);
 mvarchar 
----------
 as
(1 row)

select 'asd '::mchar(5)::mvarchar(2);
 mvarchar 
----------
 as
(1 row)

select 'asd'::mchar::mvarchar(5);
 mvarchar 
----------
 asd
(1 row)

select 'asd '::mchar::mvarchar(5);
 mvarchar 
----------
 asd
(1 row)

select 'asd'::mchar(2)::mvarchar(5);
 mvarchar 
----------
 as
(1 row)

select 'asd '::mchar(2)::mvarchar(5);
 mvarchar 
----------
 as
(1 row)

select 'asd'::mchar(5)::mvarchar(5);
 mvarchar 
----------
 asd  
(1 row)

select 'asd '::mchar(5)::mvarchar(5);
 mvarchar 
----------
 asd  
(1 row)

select 'asd'::mvarchar::mchar;
 mchar 
-------
 asd
(1 row)

select 'asd '::mvarchar::mchar;
 mchar 
-------
 asd
(1 row)

select 'asd'::mvarchar(2)::mchar;
 mchar 
-------
 as
(1 row)

select 'asd '::mvarchar(2)::mchar;
 mchar 
-------
 as
(1 row)

select 'asd'::mvarchar(5)::mchar;
 mchar 
-------
 asd
(1 row)

select 'asd '::mvarchar(5)::mchar;
 mchar 
-------
 asd
(1 row)

select 'asd'::mvarchar::mchar(2);
 mchar 
-------
 as
(1 row)

select 'asd '::mvarchar::mchar(2);
 mchar 
-------
 as
(1 row)

select 'asd'::mvarchar(2)::mchar(2);
 mchar 
-------
 as
(1 row)

select 'asd '::mvarchar(2)::mchar(2);
 mchar 
-------
 as
(1 row)

select 'asd'::mvarchar(5)::mchar(2);
 mchar 
-------
 as
(1 row)

select 'asd '::mvarchar(5)::mchar(2);
 mchar 
-------
 as
(1 row)

select 'asd'::mvarchar::mchar(5);
 mchar 
-------
 asd  
(1 row)

select 'asd '::mvarchar::mchar(5);
 mchar 
-------
 asd  
(1 row)

select 'asd'::mvarchar(2)::mchar(5);
 mchar 
-------
 as   
(1 row)

select 'asd '::mvarchar(2)::mchar(5);
 mchar 
-------
 as   
(1 row)

select 'asd'::mvarchar(5)::mchar(5);
 mchar 
-------
 asd  
(1 row)

select 'asd '::mvarchar(5)::mchar(5);
 mchar 
-------
 asd  
(1 row)

select 'asd'::mchar || '123';
 ?column? 
----------
 asd123
(1 row)

select 'asd'::mchar || '123'::mchar;
 ?column? 
----------
 asd123
(1 row)

select 'asd'::mchar || '123'::mvarchar;
 ?column? 
----------
 asd123
(1 row)

select 'asd '::mchar || '123';
 ?column? 
----------
 asd123
(1 row)

select 'asd '::mchar || '123'::mchar;
 ?column? 
----------
 asd123
(1 row)

select 'asd '::mchar || '123'::mvarchar;
 ?column? 
----------
 asd123
(1 row)

select 'asd '::mchar || '123 ';
 ?column? 
----------
 asd123
(1 row)

select 'asd '::mchar || '123 '::mchar;
 ?column? 
----------
 asd123
(1 row)

select 'asd '::mchar || '123 '::mvarchar;
 ?column? 
----------
 asd123 
(1 row)

select 'asd'::mvarchar || '123';
 ?column? 
----------
 asd123
(1 row)

select 'asd'::mvarchar || '123'::mchar;
 ?column? 
----------
 asd123
(1 row)

select 'asd'::mvarchar || '123'::mvarchar;
 ?column? 
----------
 asd123
(1 row)

select 'asd '::mvarchar || '123';
 ?column? 
----------
 asd 123
(1 row)

select 'asd '::mvarchar || '123'::mchar;
 ?column? 
----------
 asd 123
(1 row)

select 'asd '::mvarchar || '123'::mvarchar;
 ?column? 
----------
 asd 123
(1 row)

select 'asd '::mvarchar || '123 ';
 ?column? 
----------
 asd 123 
(1 row)

select 'asd '::mvarchar || '123 '::mchar;
 ?column? 
----------
 asd 123
(1 row)

select 'asd '::mvarchar || '123 '::mvarchar;
 ?column? 
----------
 asd 123 
(1 row)

select 'asd'::mchar(2) || '123';
 ?column? 
----------
 as123
(1 row)

select 'asd'::mchar(2) || '123'::mchar;
 ?column? 
----------
 as123
(1 row)

select 'asd'::mchar(2) || '123'::mvarchar;
 ?column? 
----------
 as123
(1 row)

select 'asd '::mchar(2) || '123';
 ?column? 
----------
 as123
(1 row)

select 'asd '::mchar(2) || '123'::mchar;
 ?column? 
----------
 as123
(1 row)

select 'asd '::mchar(2) || '123'::mvarchar;
 ?column? 
----------
 as123
(1 row)

select 'asd '::mchar(2) || '123 ';
 ?column? 
----------
 as123
(1 row)

select 'asd '::mchar(2) || '123 '::mchar;
 ?column? 
----------
 as123
(1 row)

select 'asd '::mchar(2) || '123 '::mvarchar;
 ?column? 
----------
 as123 
(1 row)

select 'asd'::mvarchar(2) || '123';
 ?column? 
----------
 as123
(1 row)

select 'asd'::mvarchar(2) || '123'::mchar;
 ?column? 
----------
 as123
(1 row)

select 'asd'::mvarchar(2) || '123'::mvarchar;
 ?column? 
----------
 as123
(1 row)

select 'asd '::mvarchar(2) || '123';
 ?column? 
----------
 as123
(1 row)

select 'asd '::mvarchar(2) || '123'::mchar;
 ?column? 
----------
 as123
(1 row)

select 'asd '::mvarchar(2) || '123'::mvarchar;
 ?column? 
----------
 as123
(1 row)

select 'asd '::mvarchar(2) || '123 ';
 ?column? 
----------
 as123 
(1 row)

select 'asd '::mvarchar(2) || '123 '::mchar;
 ?column? 
----------
 as123
(1 row)

select 'asd '::mvarchar(2) || '123 '::mvarchar;
 ?column? 
----------
 as123 
(1 row)

select 'asd'::mchar(4) || '143';
 ?column? 
----------
 asd 143
(1 row)

select 'asd'::mchar(4) || '123'::mchar;
 ?column? 
----------
 asd 123
(1 row)

select 'asd'::mchar(4) || '123'::mvarchar;
 ?column? 
----------
 asd 123
(1 row)

select 'asd '::mchar(4) || '123';
 ?column? 
----------
 asd 123
(1 row)

select 'asd '::mchar(4) || '123'::mchar;
 ?column? 
----------
 asd 123
(1 row)

select 'asd '::mchar(4) || '123'::mvarchar;
 ?column? 
----------
 asd 123
(1 row)

select 'asd '::mchar(4) || '123 ';
 ?column? 
----------
 asd 123
(1 row)

select 'asd '::mchar(4) || '123 '::mchar;
 ?column? 
----------
 asd 123
(1 row)

select 'asd '::mchar(4) || '123 '::mvarchar;
 ?column? 
----------
 asd 123 
(1 row)

select 'asd'::mvarchar(4) || '123';
 ?column? 
----------
 asd123
(1 row)

select 'asd'::mvarchar(4) || '123'::mchar;
 ?column? 
----------
 asd123
(1 row)

select 'asd'::mvarchar(4) || '123'::mvarchar;
 ?column? 
----------
 asd123
(1 row)

select 'asd '::mvarchar(4) || '123';
 ?column? 
----------
 asd 123
(1 row)

select 'asd '::mvarchar(4) || '123'::mchar;
 ?column? 
----------
 asd 123
(1 row)

select 'asd '::mvarchar(4) || '123'::mvarchar;
 ?column? 
----------
 asd 123
(1 row)

select 'asd '::mvarchar(4) || '123 ';
 ?column? 
----------
 asd 123 
(1 row)

select 'asd '::mvarchar(4) || '123 '::mchar;
 ?column? 
----------
 asd 123
(1 row)

select 'asd '::mvarchar(4) || '123 '::mvarchar;
 ?column? 
----------
 asd 123 
(1 row)

select 'asd '::mvarchar(4) || '123 '::mchar(4);
 ?column? 
----------
 asd 123 
(1 row)

select 'asd '::mvarchar(4) || '123 '::mvarchar(4);
 ?column? 
----------
 asd 123 
(1 row)

select 'asd '::mvarchar(4) || '123'::mchar(4);
 ?column? 
----------
 asd 123 
(1 row)

select 'asd '::mvarchar(4) || '123'::mvarchar(4);
 ?column? 
----------
 asd 123
(1 row)

select 1 where 'f'::mchar='F'::mvarchar;
 ?column? 
----------
        1
(1 row)

select 1 where 'f'::mchar='F '::mvarchar;
 ?column? 
----------
        1
(1 row)

select 1 where 'f '::mchar='F'::mvarchar;
 ?column? 
----------
        1
(1 row)

select 1 where 'f '::mchar='F '::mvarchar;
 ?column? 
----------
        1
(1 row)

select 1 where 'f'::mchar='F'::mvarchar(2);
 ?column? 
----------
        1
(1 row)

select 1 where 'f'::mchar='F '::mvarchar(2);
 ?column? 
----------
        1
(1 row)

select 1 where 'f '::mchar='F'::mvarchar(2);
 ?column? 
----------
        1
(1 row)

select 1 where 'f '::mchar='F '::mvarchar(2);
 ?column? 
----------
        1
(1 row)

select 1 where 'f'::mchar(2)='F'::mvarchar;
 ?column? 
----------
        1
(1 row)

select 1 where 'f'::mchar(2)='F '::mvarchar;
 ?column? 
----------
        1
(1 row)

select 1 where 'f '::mchar(2)='F'::mvarchar;
 ?column? 
----------
        1
(1 row)

select 1 where 'f '::mchar(2)='F '::mvarchar;
 ?column? 
----------
        1
(1 row)

select 1 where 'f'::mchar(2)='F'::mvarchar(2);
 ?column? 
----------
        1
(1 row)

select 1 where 'f'::mchar(2)='F '::mvarchar(2);
 ?column? 
----------
        1
(1 row)

select 1 where 'f '::mchar(2)='F'::mvarchar(2);
 ?column? 
----------
        1
(1 row)

select 1 where 'f '::mchar(2)='F '::mvarchar(2);
 ?column? 
----------
        1
(1 row)

select 1 where 'foo'::mchar='FOO'::mvarchar;
 ?column? 
----------
        1
(1 row)

select 1 where 'foo'::mchar='FOO '::mvarchar;
 ?column? 
----------
        1
(1 row)

select 1 where 'foo '::mchar='FOO'::mvarchar;
 ?column? 
----------
        1
(1 row)

select 1 where 'foo '::mchar='FOO '::mvarchar;
 ?column? 
----------
        1
(1 row)

select 1 where 'foo'::mchar='FOO'::mvarchar(2);
 ?column? 
----------
(0 rows)

select 1 where 'foo'::mchar='FOO '::mvarchar(2);
 ?column? 
----------
(0 rows)

select 1 where 'foo '::mchar='FOO'::mvarchar(2);
 ?column? 
----------
(0 rows)

select 1 where 'foo '::mchar='FOO '::mvarchar(2);
 ?column? 
----------
(0 rows)

select 1 where 'foo'::mchar(2)='FOO'::mvarchar;
 ?column? 
----------
(0 rows)

select 1 where 'foo'::mchar(2)='FOO '::mvarchar;
 ?column? 
----------
(0 rows)

select 1 where 'foo '::mchar(2)='FOO'::mvarchar;
 ?column? 
----------
(0 rows)

select 1 where 'foo '::mchar(2)='FOO '::mvarchar;
 ?column? 
----------
(0 rows)

select 1 where 'foo'::mchar(2)='FOO'::mvarchar(2);
 ?column? 
----------
        1
(1 row)

select 1 where 'foo'::mchar(2)='FOO '::mvarchar(2);
 ?column? 
----------
        1
(1 row)

select 1 where 'foo '::mchar(2)='FOO'::mvarchar(2);
 ?column? 
----------
        1
(1 row)

select 1 where 'foo '::mchar(2)='FOO '::mvarchar(2);
 ?column? 
----------
        1
(1 row)

Select 'f'::mchar(1) Union Select 'o'::mvarchar(1);
 mchar 
-------
 f
 o
(2 rows)

Select 'f'::mvarchar(1) Union Select 'o'::mchar(1);
 mvarchar 
----------
 f
 o
(2 rows)

select * from chvch where ch=vch;
      ch      |    vch     
--------------+------------
 No spaces    | No spaces
 One space    | One space 
 1 space      | 1 space 
(3 rows)

select ch.* from ch, (select 'dEfg'::mvarchar as q) as p  where  chcol > p.q;
              chcol               
----------------------------------
 ee                              
 Ee                              
(2 rows)

create index qq on ch (chcol);
set enable_seqscan=off;
select ch.* from ch, (select 'dEfg'::mvarchar as q) as p  where  chcol > p.q;
              chcol               
----------------------------------
 ee                              
 Ee                              
(2 rows)

set enable_seqscan=on;
--\copy chvch to 'results/chvch.dump' binary
--truncate table chvch;
--\copy chvch from 'results/chvch.dump' binary
--test joins
CREATE TABLE a (mchar2 MCHAR(2) NOT NULL);
CREATE TABLE c (mvarchar255 mvarchar NOT NULL);
SELECT * FROM a, c WHERE mchar2 = mvarchar255;
 mchar2 | mvarchar255 
--------+-------------
(0 rows)

SELECT * FROM a, c WHERE mvarchar255 = mchar2;
 mchar2 | mvarchar255 
--------+-------------
(0 rows)

DROP TABLE a;
DROP TABLE c;
select * from (values
    ('е'::mchar),('ё'),('еа'),('еб'),('ее'),('еж'),('ёа'),('ёб'),('ёё'),('ёж'),('ёе'),('её'))
    z order by 1;
 column1 
---------
 е
 ё
 еа
 ёа
 еб
 ёб
 ее
 её
 ёе
 ёё
 еж
 ёж
(12 rows)

select 'ё'::mchar = 'е';
 ?column? 
----------
 f
(1 row)

select 'Ё'::mchar = 'Е';
 ?column? 
----------
 f
(1 row)

select 'й'::mchar = 'и';
 ?column? 
----------
 f
(1 row)

select 'Й'::mchar = 'И';
 ?column? 
----------
 f
(1 row)

select mvarchar_icase_cmp('ёа','еб'), mvarchar_icase_cmp('еб','ё'),
    mvarchar_icase_cmp('ё', 'ёа');
 mvarchar_icase_cmp | mvarchar_icase_cmp | mvarchar_icase_cmp 
--------------------+--------------------+--------------------
                 -1 |                  1 |                 -1
(1 row)

